hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java

```diff
@@ -141,8 +141,11 @@ public synchronized void run() {
             if (LOG.isDebugEnabled()) {
               LOG.debug("task execution result size {}", result.getSize());
             }
-          } catch (Exception e) {
-            LOG.warn("Background task execution failed", e);
+          } catch (Throwable e) {
+            LOG.error("Background task execution failed", e);
+            if (e instanceof Error) {
+              throw e;
```

Comment:
Sorry, this is the right syntax:

```suggestion
              throw (Error) e;
```